### ISSUE TITLE
Managed Registry in SBG

### ISSUE THREAD

**Issue by danvy** [ovh/public-cloud-roadmap#237]
Managed Registry is available in GRA only making PRA setup more complex than expected.
Hosting the registry in a VM just for the PRA would increase risk in this not ideal situation.
BHS and US support are in the backlog, although there would be a sovereignty issue.

**Comment by mhurtrel** [comment]
Hi @danvy  and thanks for raising this usecase.
As you mentionned, our short/mid term priority is to have the service available in each continent. At the same time, we work on exciting new features for manage dprivate regitry, including upgrade to a very recent Harbor version.

We currently have a small number of customers actively looking to start a service in multiple regions, though indeed PRA makes sense, and we plan to advertise this when we will have the service in Canada. We see less concerns for having this service in Canada (as a PRA for example) because this usually contains data less sensible to soverigntly (comapred to managed kubernets, containing logs of you web applications for example).

We may prioritize, after canada, another european region (german one for example). Would it fit your use case ? Or you abdoslutely need a second FR region ? Note that harbor comes with a replication feature that is compatibile with most registries on the market, if this can help in the meantime.

**Comment by danvy** [comment]
> including upgrade to a very recent Harbor version.
This makes me thoughtful. Isn't it one of the very basic value prop of a PaaS offer?

> We currently have a small number of customers actively looking to start a service in multiple regions, though indeed PRA makes sense, and we plan to advertise this when we will have the service in Canada. We see less concerns for having this service in Canada (as a PRA for example) because this usually contains data less sensible to soverigntly (compared to managed kubernets, containing logs of you web applications for example).
- Not having an alternative DC is a show stopper for me. I can detail why if needed. Canada or US would make it for my need of a dev/test registry. Germany would be far better. This would greatly simply the communication ("All in Europe" vs "Nearly all expect this and that").
- What about IP?

> We may prioritize, after canada, another european region (german one for example). Would it fit your use case ? Or you abdoslutely need a second FR region ? Note that harbor comes with a replication feature that is compatibile with most registries on the market, if this can help in the meantime.
Having our production infrastructure spread in more than one region would induce some risk mitigation which is far from an ideal scenario. There no problem for dev/test.

**Comment by mhurtrel** [comment]
Thanks for your detailed feedback.

While we make sure we patch any known CVE, we are indeed a bit late in following major new Harbor versions. We we focused onn keeping up with stong growth and will catch-up to a very recent version soon (may not be the latest one, as we do like, like most cloud actors and enterpise software editor, to have new versions get battle tested before managing thousands in production.)

We will consider accelerating on a second european region after we open Canada. Note that this specific servuce is offered trough a dedicated subdomain and no dedicated IP.

**Comment by LisaPerrier** [comment]
Hello, 
Thanks to your feedback, Managed Private Registry in SBG has been prioritized on our end for the end of the year 2023 :)

**Comment by mhurtrel** [comment]
@danvy keep it mind that it is already avalailable in 2 regions (canada has been opened this week) and will be in germany (limburg) next month

**Comment by danvy** [comment]
> @danvy keep it mind that it is already avalailable in 2 regions (canada has been opened this week) and will be in germany (limburg) next month

Yep, I have those information in mind. Having a unit of infrastructure spread in different location will increase risk where we need to lower it. When Germany will be available (The console usually says Frankfurt DE, as Limburg is not so far, I think we're talking about the same place) a full German deployment will be an option for those with EU requirements. For French requirements, they'll have to wait until EOY.

**Comment by anthonyroussel** [comment]
Hello, any update on Managed Private Registry in SBG?

It was prioritised for end of 2023. As of today, this service still does not appear to be available in the region, and it remains important for Disaster-Recovery architecture. Having a local registry is necessary for maintaining strict fault domain isolation and avoiding cross-region dependencies that could impact our recovery strategy.
